### DISCUSSION TITLE
Moment.js support

### DISCUSSION THREAD

**Discussion by samg666** [Budibase/budibase#4498] [General]
Hello,

It seems currently Budibase JavaScript don't support Moment.js, or maybe I didn't find a right way to use Moment.js. Can anyone share idea how to use Moment.js on Budibase? Or if it's not supported, is there any way to get this supported as I have many days needs to be operate.

**Comment by mjashanks** [comment]
We don't support this at all right now. It would be nice to allow importing of scripts in general.

When we use Javascript bindings, the javascript is completely sandboxed - so even if you import `moment` via en `Embed`, the Javascript bindings wont see the import.

**Comment by kerwal** [comment]
(My apologies for resurrecting)
I found that copy/pasting the contents of moment.min.js at the top of the script will make moment() available to use. It's practically the same thing that importing to the Node VM will do. It's just a tad messy.

[https://raw.githubusercontent.com/moment/moment/develop/min/moment.min.js](https://raw.githubusercontent.com/moment/moment/develop/min/moment.min.js)

Cheers!

**Comment by Topener** [comment]
Thanks for the suggestion! As @mjashanks mentioned we do not have support for importing scripts yet. Your workaround is one that's fine for now but of course not a recommended approach.

We just added a ticket to our roadmap to add DayJS support: #4564 

MomentJS is considered, [by MomentJS](https://momentjs.com/docs/#/-project-status/), as a legacy framework and no longer recommended/updated by them. They've listed several alternatives including DayJS. 

When we add importing scripts functionality you can of course always add MomentJS, but I recommend monitoring the DayJS ticket so you can get updated when it's added!
